pkg/postage/service.go

```diff
@@ -44,58 +41,26 @@ type Service interface {
 // stores the active batches.
 type service struct {
 	lock         sync.Mutex
-	store        storage.StateStorer
+	store        storage.Store
 	postageStore Storer
 	chainID      int64
-	issuers      []*StampIssuer
 }
 
 // NewService constructs a new Service.
-func NewService(store storage.StateStorer, postageStore Storer, chainID int64) (Service, error) {
-	s := &service{
+func NewService(store storage.Store, postageStore Storer, chainID int64) Service {
+	return &service{
 		store:        store,
 		postageStore: postageStore,
 		chainID:      chainID,
 	}
-	if err := s.store.Iterate(s.key(), func(_, value []byte) (bool, error) {
-		st := &StampIssuer{}
-		if err := st.UnmarshalBinary(value); err != nil {
-			return false, err
-		}
-		_ = s.add(st)
-		return false, nil
-	}); err != nil {
-		return nil, err
-	}
-
-	return s, nil
 }
 
-// Add adds a stamp issuer to the active issuers.
+// Add adds a stamp issuer to stamperstore.
 func (ps *service) Add(st *StampIssuer) error {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 
-	if ps.add(st) {
-		if err := ps.store.Put(ps.keyForIndex(st.data.BatchID), st); err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// add adds a stamp issuer to the active issuers and returns false if it is already present.
-func (ps *service) add(st *StampIssuer) bool {
-
-	for _, v := range ps.issuers {
-		if bytes.Equal(st.data.BatchID, v.data.BatchID) {
-			return false
-		}
-	}
-	ps.issuers = append(ps.issuers, st)
-
-	return true
+	return ps.save(st)
```

Comment:
Here we need to check if the stamp issuer is already saved. Otherwise this will overwrite the stampIssuer.Data

Reply:
This still needs to be resolved right?

Add can be called from 2 places, so if we already have the stamp issuer created we should not update it. For example, the first time Add is called is when the node creates a batch. Then, it can receive a HandleCreate event as well which may be delayed or could have got on restart or resync of the node. If we had already issued some indexes then the update will overwrite them.